app/src/main/java/com/epfl/drawyourpath/database/Database.kt

```diff
@@ -245,15 +253,77 @@ abstract class Database {
     ): CompletableFuture<Unit>
 
     /**
-     * Function used to retrieve the tournament corresponding to the ID.
-     * @param tournamentId the id of the tournaments to retrieve
+     * Function used to retrieve the tournament corresponding to the ID. USe only if you need to get
+     * everything about a tournament (including posts and participants). Otherwise, you should use
+     * [getTournamentInfo], [getTournamentPosts] or [getTournamentParticipantsId].
+     * @param tournamentId the id of the tournaments to retrieve.
      * @return a future that contains the tournament if successful, or an exception if an error
      * occurred with the db.
      */
     abstract fun getTournament(
         tournamentId: String,
     ): CompletableFuture<Tournament>
 
+    /**
+     * Function used to retrieve all posts from the tournament corresponding to the ID.
+     * @param tournamentId the id of the tournaments to retrieve.
+     * @return a future that contains the list of the posts of the tournament.
+     */
+    abstract fun getTournamentPosts(
+        tournamentId: String,
+    ): CompletableFuture<List<TournamentPost>>
+
+    /**
+     * Function used to retrieve the participants of a tournament.
+     * @param tournamentId the id of the tournaments from which we want the participants.
+     * @return a future that contains the list of participants if successful.
+     */
+    abstract fun getTournamentParticipantsId(
+        tournamentId: String,
+    ): CompletableFuture<List<String>>
```

Comment:
Same here you. shoudl use return live datatype like for the getFriendsList()function